### PR TITLE
Cyborg self-repair nerf + QoL changes

### PR DESCRIPTION
**Pull request by Rukofamicom**
## About The Pull Request

1) Cyborgs are no longer able to repair themselves
2) All cyborgs that did not previously have a welding tool have a 10u emergency tool now
3) Repairing a cyborg is now repeated in steps until completion or canceled. No more click spamming
4) As a result of 3, repairing a cyborg now takes time instead of just spam - 10 damage takes 6 seconds. This is about on par with some of the best healing medicine for fleshy players, but with none of the chemistry involvement.  

## Why It's Good For The Game

~~valid-happy cyborg bad~~

The ability to rapidly heal, especially self-heal, is a bit too much on top of all the rest of the combat advantages cyborgs have, even when factoring the couple of extreme weaknesses they have. I also believe the primary reason the station sees such an abundance of engineering cyborgs is their access to self healing. Adding a lesser welding tool to the rest of the cyborgs should further enable the other modules without taking anything away from engi borgs specifically. 

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Cannot repair self
image

Repair steps progressing automatically. Also checked wiring. 
image

Clown module showing off his welding tool
image

</details>

## Changelog
:cl:
balance: Cyborgs are no longer able to repair themselves and must seek assistance from others
tweak: All cyborg modules now have access to a welding tool
tweak: Cyborg repairs are now repeated automatically without the need to spam click
/:cl:

### CODE REVIEW

**Comment by Rukofamicom** [comment]
One of the two most important things supplied by #6759 / #7347 but atomized out of them.

**Comment by PowerfulBacon** [comment]
I like these changes, haven't looked at codw

**Comment by WrillWasTaken** [comment]
How will this impact mining cyborgs?

**Comment by Rukofamicom** [comment]
Mining cyborgs will need outside help for repairs just like any other would.

**Comment by ro5490** [comment]
I think the part fabricator can make auto-repair upgrades for mechs, not sure if it can for cyborgs but if not that might be a good idea to look at, for a later-round upgrade, maybe limited to mining borgs only though? miners and mining borgs tend to get damaged real quick if they slip up at their jobs, and almost always go solo

**Comment by Rukofamicom** [comment]
There's already an auto-repair module available for all cyborgs. 
It's all that great unless you have a Xenobio power cell though.

**Review by PowerfulBacon — DISMISSED** [review]

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -429,41 +429,43 @@
...
 		var/obj/item/stack/cable_coil/coil = W
-		if (getFireLoss() > 0 || getToxLoss() > 0)
-			if(src == user)
-				to_chat(user, "<span class='notice'>You start fixing yourself.</span>")
-				if(!do_after(user, 50, target = src))
-					return
-			if (coil.use(1))
-				adjustFireLoss(-30)
-				adjustToxLoss(-30)
+		while(getFireLoss() && do_after(user, 60, target = src))
```
Comment (itsmeow) [inline code/modules/mob/living/silicon/robot/robot.dm]: what about toxloss?

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/silicon/robot/robot.dm (hunk 1 above)
Comment (Rukofamicom) [reply code/modules/mob/living/silicon/robot/robot.dm]: Thanks much, I missed that when touching up my old code

**Review by PowerfulBacon — APPROVED** [review]

**Review by itsmeow — APPROVED** [review]

**Comment by Fronsis** [comment]
> There's already an auto-repair module available for all cyborgs. It's all that great unless you have a Xenobio power cell though.

I'm honestly lost in memory ever since you proposed the first PR of borgs overhaul so i'm unsure if this was merged or got left behind on the main PR:

Was the self-repair module rate of repairing buffed? or that got left behind on the main PR? To compensate the loss of self-repairing and encourage the borgs to visit robotics.

Considering Bacon and others have voiced(on discord recently) a consideration to reduce the combat capabilities of non-combat oriented borgs to avoid them going gamer mode perhaps the slow-down thing could be brought up again and see what kind of discussion we can have

**Comment by Rukofamicom** [comment]
My original PR buffed the self-repair module, as well as changing many other things with a scope that was too wide for its own good.

* The other PR also added damage slowdown
* Lessened the weakness to flashes
* Removed flashes from most cyborgs
* Changed the kind of flashes cyborgs were made with by adding a new one
* Greatly reduced cyborg weakness to laser pointer
* Weakened cyborg hacked modules since they were harder to instantly stop
* Gave cyborg hyposprays a use time

~~It was one of my first major PRs and I didn't have the self control to not implement any vaguely related idea I had.~~

**Comment by Fronsis** [comment]
I see!

Seeing as this was merged and from the glimpses of conversation i saw on Discord from Bacon, some admins and other players there might be shot at most of these stuff on their own if you ever find time, guess we'll have to see how the borg meta plays out with only this being merged, i feel like miner borgs will probably suffer the most(not counting the gamer engi borgs) in terms of repairing themselves, that's why i think a follow up to QoL would be the buff to the self-repair module

Also unrelated but i learned that you could upgrade laser pointers with t4 to guarantee the stuns on borgs lol

**Comment by Rukofamicom** [comment]
> Also unrelated but i learned that you could upgrade laser pointers with t4 to guarantee the stuns on borgs lol

From quite a long range!

But yeah, I'm hoping to go back over both prior versions and atomize the changes out, probably along these lines:

1) Improve repair module + damage slowdown 
2) Reduce weakness to flash/laser + adjust hypospray and hacked modules

The removal of flashes from all but peacekeeper and standard cyborg probably won't happen unless maintainer requested. If flashes are too strong, that's something that needs to be addressed with flashes and/or whatever is being shut down too hard by them, not cyborgs.

**Pull request merged**